### PR TITLE
feat: Extract block_reverter into separate crate

### PR DESCRIPTION
**Pull request by AnastasiiaVashchuk**
## What ❔
Extract `zksync_core::block_reverter` to separate the crate in the `node` folder

## Why ❔

Part of the to "modularize the codebase" process, required for legolizer and publishing on crates.io.

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.
- [ ] Spellcheck has been run via `zk spellcheck`.
- [ ] Linkcheck has been run via `zk linkcheck`.

### CODE REVIEW

**Review by montekki — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -221,3 +221,4 @@ zksync_crypto_primitives = { path = "core/lib/crypto_primitives" }
 zksync_node_framework = { path = "core/node/node_framework" }
 zksync_eth_watch = { path = "core/node/eth_watch" }
 zksync_shared_metrics = { path = "core/node/shared_metrics" }
+zksync_block_reverter = { path = "core/node/block_reverter"}
```
Comment (slowli) [inline Cargo.toml:224]: Same as #1636, you should include this crate in `workspace.members`.

**Review by slowli — COMMENTED** [review]
Wonder how the reverter will fit into the node framework (it doesn't look like a component), but if it does, I'm all for it.

File: Cargo.toml (hunk 1 above)
Comment (AnastasiiaVashchuk) [reply Cargo.toml:224]: Could u pls explain why we need to include this into `workspace.members`?

**Comment by AnastasiiaVashchuk** [comment]
>Wonder how the reverter will fit into the node framework (it doesn't look like a component), but if it does, I'm all for it.

yeah,.. it's more like a tool..but it lives in `zksync_core` rn + Igor mentioned it when we discussed modules' extraction with him.

**Review by slowli — DISMISSED** [review]

**Review by slowli — APPROVED** [review]

**Pull request merged**